### PR TITLE
fix part of the windows tests

### PR DESCRIPTION
**Pull request by partouf**

### CODE REVIEW

File: test/base-compiler-tests.ts [hunk 1]
```diff
@@ -658,12 +658,18 @@ Args: []
         expect(() => compiler.getExtraFilepath('/tmp/somefolder', '../test.h')).toThrow(Error);
         expect(() => compiler.getExtraFilepath('/tmp/somefolder', './../test.h')).toThrow(Error);
 
-        expect(compiler.getExtraFilepath('/tmp/somefolder', '/tmp/someotherfolder/test.h')).toEqual(
-            '/tmp/somefolder/tmp/someotherfolder/test.h',
-        );
+        if (process.platform === 'win32') {
+            expect(compiler.getExtraFilepath('/tmp/somefolder', '/tmp/someotherfolder/test.h')).toEqual(
+                '\\tmp\\somefolder\\tmp\\someotherfolder\\test.h',
```
Comment (mattgodbolt) [inline test/base-compiler-tests.ts:663]: We could:
```ts
function normalizedFilepath(path) {
  if (is windows) return path.replace('/', '\\');
return path;
}
```
and then we don't need to add all the `if`s around here?

File: test/exec-tests.ts [hunk 2]
```diff
@@ -367,18 +367,33 @@ describe('Execution tests', async () => {
...
-            ]);
+            if (process.platform === 'win32') {
+                expect(args).toEqual([
+                    '--config',
+                    'etc/nsjail/sandbox.cfg',
+                    '--cwd',
+                    '/app',
+                    '--bindmount',
+                    '/tmp/hellow:/app',
+                    '--env=LD_LIBRARY_PATH=/usr/lib;/app/lib',
```
Comment (mattgodbolt) [inline test/exec-tests.ts:378]: took me a while to spot the different here; for my poor old eye maybe:
```ts
const path_sep = windows ? ':' : ';';
expect(args).toEqual(...
'--env=LD_LIBRARY_PATH=/usr/lib:${path_sep}/app/lib
```

or something?

File: test/library-tests.ts [hunk 3]
```diff
@@ -128,7 +128,12 @@ describe('Library directories (c++)', () => {
             undefined,
             '/tmp/compiler-explorer-compiler-123',
         );
-        expect(qtpaths).toContain('-L/tmp/compiler-explorer-compiler-123/qt/lib');
+
+        if (process.platform === 'win32') {
+            expect(qtpaths).toContain('-L\\tmp\\compiler-explorer-compiler-123\\qt\\lib');
```
Comment (mattgodbolt) [inline test/library-tests.ts:133]: (ditto all these could use the normalisePath thing if we made a test-only thing:
```expect(normalisePath(qtpaths)).toContain(/unix/path/)```
or maybe it'd need to handle lists, or we could do a fairly straightforward mapping thing

**Review by mattgodbolt — APPROVED** [review]
some suggestions to minimise the amount of `if`s we need in the codebase and to give my poor eyesight a helping hand with `;` vs `:` 😀

**Comment by partouf** [comment]
it's difficult to figure out what's going on with the approvals diffs, cannot figure out how to make vitest show a proper diff

i'll merge this for now as the other issues are fixed

**Pull request merged**